src/xo-app/new-vm/index.js

```diff
@@ -361,7 +361,7 @@ export default class NewVm extends BaseComponent {
         value = isArray(value) ? [ ...value ] : { ...value }
         let eventValue = getEventValue(event)
         eventValue = targetObjectProp ? eventValue[targetObjectProp] : eventValue
-        stateObjectProp ? value[index][stateObjectProp] = eventValue : value[index] = eventValue
+        value[index] && stateObjectProp ? value[index][stateObjectProp] = eventValue : value[index] = eventValue
```

Comment:
Indeed, the main feature of the ternary operator over `if-else` is to be an expression.